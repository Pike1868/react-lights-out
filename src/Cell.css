.Cell {
  background-color: rgb(95, 130, 92);
  border: 1px solid rgba(255, 253, 253, 0.837);
  border-radius: 1%;
  width: 100%;
  height: 100%;
  max-height: 100px;
  text-align: center;
  font-weight: 700;
  font-size: 1.5em;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
}

.Cell-lit {
  background-color: rgba(252, 227, 64, 0.897);
}
