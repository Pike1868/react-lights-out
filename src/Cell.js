import React from "react";
import "./Cell.css";

const Cell = ({ isLit, onClick }) => {
  return (
    <div className={`Cell ${isLit ? "Cell-lit" : ""}`} onClick={onClick}></div>
  );
};

export default Cell;
