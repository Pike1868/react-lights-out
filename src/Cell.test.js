import React from "react";
import { render } from "@testing-library/react";
import Cell from "./Cell";

//Smoke Test
test("That Cell component will render without crashing", () => {
  render(<Cell isLit={false} onClick={() => console.log("clicked")} />);
});

//Snapshot Test for "off" cell
test("That correct class is applied when isLit = false", () => {
  const { asFragment } = render(<Cell isLit={false} />);
  console.log(asFragment)
  expect(asFragment()).toMatchSnapshot();
});

//Snapshot Test for "on" cell
test("correct class is applied when isLit = true", () => {
  const { asFragment } = render(<Cell isLit={true} />);

  expect(asFragment()).toMatchSnapshot();
});
