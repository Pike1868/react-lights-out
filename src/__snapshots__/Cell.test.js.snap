// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`That correct class is applied when isLit = false 1`] = `
<DocumentFragment>
  <div
    class="Cell "
  />
</DocumentFragment>
`;

exports[`correct class is applied when isLit = false 1`] = `
<DocumentFragment>
  <div
    class="Cell "
  />
</DocumentFragment>
`;

exports[`correct class is applied when isLit = true 1`] = `
<DocumentFragment>
  <div
    class="Cell Cell-lit"
  />
</DocumentFragment>
`;
